mvcc/backend/backend.go

```diff
@@ -165,31 +166,40 @@ func (b *backend) ForceCommit() {
 }
 
 func (b *backend) Snapshot() Snapshot {
+	b.batchTx.Commit()
+
+	b.mu.RLock()
+	defer b.mu.RUnlock()
+	tx, err := b.db.Begin(false)
+	if err != nil {
+		plog.Fatalf("cannot begin tx (%s)", err)
+	}
+
 	stopc, donec := make(chan struct{}), make(chan struct{})
+	dbBytes := tx.Size()
 	go func() {
 		defer close(donec)
+		// snapshotWarningTimeout is based on transferring snapshot data over a 1 gigabit/s connection
+		// with max tcp throughput of 949.285 Mbit/s limited by TCP overhead.
+		var throughputBytes int64 = 118660625 // 949.285 Mbit/s = 118660625 bytes/s
```

Comment:
i do not feel we need this accurate.

just do 100 * 1024 * 1024 // 100MB to be more conservative.